src/include/mallocMC/distributionPolicies/XMallocSIMD_impl.hpp

```diff
@@ -76,19 +75,19 @@ namespace DistributionPolicies{
  * default-struct < template-struct < command-line parameter
  */
 #ifndef MALLOCMC_DP_XMALLOCSIMD_PAGESIZE
-#define MALLOCMC_DP_XMALLOCSIMD_PAGESIZE Properties::pagesize::value
+#define MALLOCMC_DP_XMALLOCSIMD_PAGESIZE (Properties::pagesize)
 #endif
-      BOOST_STATIC_CONSTEXPR uint32 pagesize      = MALLOCMC_DP_XMALLOCSIMD_PAGESIZE;
+      static constexpr uint32 pagesize      = MALLOCMC_DP_XMALLOCSIMD_PAGESIZE;
 
       //all the properties must be unsigned integers > 0
-      BOOST_STATIC_ASSERT(!std::numeric_limits<typename Properties::pagesize::type>::is_signed);
+      static_assert(!std::numeric_limits<decltype(Properties::pagesize)>::is_signed, ""); // TODO(bgruber): I think this is too strict. std::is_convertible_v should be enough
```

Comment:
What kind of advantages do we have if we relax it and use `std::is_convertible_v`

Reply:
The advantage is that 4096 is a valid page size, not only 4096u. See e.g. the following configuration:
```
struct ScatterConfig{
  static constexpr auto pagesize = 4096u;
  static constexpr auto accessblocks = 8;
  static constexpr auto regionsize = 16;
  static constexpr auto wastefactor = 2;
  static constexpr auto resetfreedpages = false;
};
```
Only pagesize needs to be unsiged in order to compile. I think this is a usability bug.

Reply:
Shouldn't we make them all unsigned instead? :D

Reply:
I am fine with changing it to signed but we should take care that we avoid as good as possible implicit casts, even if the code is currently not checked to avoid it.

Reply:
I still don't think that this is a problem, since the class member to which the signed value is assigned to, is still unsigned (see line 80):
```
static constexpr uint32 pagesize      = MALLOCMC_DP_XMALLOCSIMD_PAGESIZE;
```

Reply:
@ax3l: in principal, all the values are semantically unsigned. But inside the policy types I just want to carry literals. So any data type is fine for me, as long as it convertes to what I expect in the implementation later. Otherwise it is a configuration error, which the compiler tells you ;)
And the implementation seems to use uint32_t internally.
About signed ints: they benefit from UB wrt. wrapping behavior so the optimizer can make more assumptions, which can help.